File: Tests/DuetCLITests/Resources/mini-dual/src-ios/Sources/GreeterFeature/GreeterFeature.swift
// Mini-tree source: the Swift half of the dual-declared greeter feature.
// The declaration-parity check extracts these names and compares them with
// the Kotlin twin's — tests break one side and assert the named violation.

public struct GreeterState: Equatable {
  public var message: String
  public var waveCount: Int
}

public enum GreeterAction: Equatable {
  case wave
  case reset(String)
}

public enum GreeterEffectPayload: Equatable {
  case log(String)
}

File: Tests/DuetCLITests/Resources/mini-dual/src-android/feature-greeter/src/main/kotlin/com/mini/greeter/GreeterFeature.kt
// Mini-tree source: the Kotlin half of the dual-declared greeter feature.
package com.mini.greeter

data class GreeterState(
    val message: String,
    val waveCount: Int,
)

sealed interface GreeterAction {
    data object Wave : GreeterAction
    data class Reset(val message: String) : GreeterAction
}

sealed interface GreeterEffectPayload {
    data class Log(val line: String) : GreeterEffectPayload
}
